org.civicrm.reqangular/src/common/components/heatmap.component.js

```diff
@@ -0,0 +1,94 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'common/modules/components'
+], function (_, components) {
+  return components.component('heatmap', {
+    bindings: {
+      values: '<'
+    },
+    controller: heatmapController,
+    controllerAs: 'heatmap',
+    template: ['$templateCache', function ($templateCache) {
+      return $templateCache.get('components/heatmap.html');
+    }]
+  });
+
+  function heatmapController () {
+    var dayLabels = [
+      { label: 'Monday', shortLabel: 'M' },
+      { label: 'Tuesday', shortLabel: 'T' },
+      { label: 'Wednesday', shortLabel: 'W' },
+      { label: 'Thursday', shortLabel: 'T' },
+      { label: 'Friday', shortLabel: 'F' },
+      { label: 'Saturday', shortLabel: 'S' },
+      { label: 'Sunday', shortLabel: 'S' }
+    ];
+    var vm = this;
+
+    vm.days = [];
+
+    vm.$onChanges = $onChanges;
+
+    /**
+     * Implementes the $onChanges method for angular controllers. When the
+     * values binding is ready, it maps the values to the days and heat ranges.
+     */
+    function $onChanges () {
+      if (vm.values) {
+        mapValuesToDaysAndHeatRanges();
+      }
+    }
+
+    /**
+     * Transform a list of week day values into another list with their
+     * heat range equivalent.
+     *
+     * The values are categorized into *low*, *medium*, *high*, and *disabled*
+     * (in case the value is a strict *false*). For the *low* and *high* range
+     * the minimum and maxim values are used. For the *medium* range the average
+     * for all the values is selected.
+     *
+     * A value of *false* is treated as *disabled*.
+     */
+    function mapValuesToDaysAndHeatRanges () {
```

Comment:
I split the code, but the calls to the functions I left them inside this function instead of `$onChanges`, in case someone calls the function directly it executes all the functions it needs to run properly.